### PR TITLE
Trev8939/raster layer geo package

### PR DESCRIPTION
**Pull request by TADraeseke**
Add a sample demonstrating use of the raster layer geo package.

### CODE REVIEW

File: java/raster-layer-geopackage/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Raster Layer GeoPackage
+
+This sample demonstrates how to open a GeoPackage, obtain a raster from the package, and display the table as a RasterLayer.
```
Comment (shellygill) [inline java/raster-layer-geopackage/README.md]: Format class names here?

File: java/raster-layer-geopackage/src/main/java/com/esri/arcgisruntime/sample/rasterlayergeopackage/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // inflate MapView from layout
+    mMapView = findViewById(R.id.mapView);
+
+    // create a map with the BasemapType topographic
```
Comment (shellygill) [inline java/raster-layer-geopackage/src/main/java/com/esri/arcgisruntime/sample/rasterlayergeopackage/MainActivity.java]: Comment about basemap type does not match code.

File: java/raster-layer-geopackage/src/main/java/com/esri/arcgisruntime/sample/rasterlayergeopackage/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    // open the GeoPackage
+    GeoPackage geoPackage = new GeoPackage(
+        Environment.getExternalStorageDirectory() + getString(R.string.geopackage_path));
+    geoPackage.loadAsync();
+    geoPackage.addDoneLoadingListener(() -> {
+      if (geoPackage.getLoadStatus() == LoadStatus.LOADED) {
+        // read raster images and get the first one
+        Raster geoPackageRaster = geoPackage.getGeoPackageRasters().get(0);
+
+        // make sure an image was found in the package
```
Comment (shellygill) [inline java/raster-layer-geopackage/src/main/java/com/esri/arcgisruntime/sample/rasterlayergeopackage/MainActivity.java]: If there's no image in the package, I'd expect an exception will already have been thrown by this point - as the raster at index 0 has already been accessed in the line above.

Comment (TADraeseke) [reply java/raster-layer-geopackage/src/main/java/com/esri/arcgisruntime/sample/rasterlayergeopackage/MainActivity.java]: Hmm, good point...

**Comment by TADraeseke** [comment]
@shellygill - Thanks! Updated as per your comments.

As per your third comment, I've added a check 
`if (!geoPackage.getGeoPackageRasters().isEmpty()) {`
before the `.get()`. Think this resolves the silly bit of code I wrote earlier.

**Pull request merged**